### PR TITLE
Issue 8: Handle the websocket close event

### LINKED ISSUE #8: Add support for websocket close event
https://forum.defold.com/t/extension-websocket-beta-testing/66216/31?u=mathias_westerdahl

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,4 +9,14 @@ We recommend using a link to a zip file of a [specific release](https://github.c
...
+
+## Debugging
+
+In order to make it easier to debug this extension, we provide a `game.project` setting `websocket.debug`.
+
+Set it to:
+
+* `0` to disable debugging (i.e. no debug output).
+* `1` to display state changes.
+* `2` to display the messages sent and received.
```
Comment (JCash) [inline README.md:22]: Good to have for the users to be able to debug what's happening.

File: docs/index.md [hunk 2]
```diff
@@ -20,10 +20,13 @@ local function websocket_callback(self, conn, data)
         print("Connected " .. conn)
         -- self.connection = conn
     elseif data.event == websocket.EVENT_ERROR then
-        print("Error:", data.error)
+        print("Error:", data.message)
```
Comment (JCash) [inline docs/index.md:23]: Api change: All events now have the "message".

File: websocket/src/websocket.cpp [hunk 3]
```diff
@@ -337,14 +339,8 @@ static void HandleCallback(WebsocketConnection* conn, int event, int msg_offset,
...
-    if (EVENT_ERROR == event) {
-        lua_pushlstring(L, conn->m_Buffer, conn->m_BufferSize);
-        lua_setfield(L, -2, "error");
-    }
-    else if (EVENT_MESSAGE == event) {
-        lua_pushlstring(L, conn->m_Buffer + msg_offset, msg_length);
-        lua_setfield(L, -2, "message");
-    }
+    lua_pushlstring(L, conn->m_Buffer + msg_offset, msg_length);
+    lua_setfield(L, -2, "message");
```
Comment (JCash) [inline websocket/src/websocket.cpp:343]: Easier to reason about (and thus remember)

File: websocket/src/wslay_callbacks.cpp [hunk 4]
```diff
@@ -94,7 +86,7 @@ ssize_t WSL_RecvCallback(wslay_event_context_ptr ctx, uint8_t *buf, size_t len,
     dmSocket::Result socket_result = Receive(conn, buf, len, &r);
 
     if (dmSocket::RESULT_OK == socket_result && r == 0)
-        socket_result = dmSocket::RESULT_WOULDBLOCK;
+        socket_result = dmSocket::RESULT_CONNABORTED;
```
Comment (JCash) [inline websocket/src/wslay_callbacks.cpp:89]: For some reason, when the server does an abnormal exit (1006), we don't get that message, and we'd just wait forever.

**Review by britzl — APPROVED** [review]
Looks good!

**Pull request merged**